docs/tutorials/discovery-fabric-network.md

```diff
@@ -0,0 +1,245 @@
+work as the application does not have visibility into the virtual system and
+the port address as seen by the peer's discovery service.
+
+Set the `asLocalhost` to true, the default is true.
+```
+await gateway.connect(connectionProfile, {discovery: { enabled: true, asLocalhost: true}});
+const network = await gateway.getNetork('mychannel');
+```
+
+Notice in the following definition of a peer from a docker-compose file.
+The port number is the same and has been defined along with the
+host name `peer0.org1.example.com:7051` for the peer and gossip settings.
+The node.js fabric-client application running outside of of the docker
+containers will use `localhost:7051`. There is a mapping of the host name to
+'localhost' but there is no mapping of the port address.
+
+NOTE: All port addresses must be unique within the docker compose configuration
+due to the mapping of the virtual host name to the `localhost`. All port addresses
+must also be mapped within the docker compose to the same external address,
+this will require that the fabric environment settings indicate to the peer and
+orderer that port address.
+
+```
+peer0.org1.example.com:
+  container_name: peer0.org1.example.com
+  image: hyperledger/fabric-peer
+  environment:
+	- CORE_VM_ENDPOINT=unix:///host/var/run/docker.sock
+	- CORE_PEER_ID=peer0.org1.example.com
+	- CORE_PEER_ADDRESS=peer0.org1.example.com:7051 // notice how this matches the port mapping below
+	- CORE_PEER_LISTENADDRESS=peer0.org1.example.com:7051
+	- CORE_PEER_GOSSIP_ENDPOINT=peer0.org1.example.com:7051
+	- CORE_PEER_GOSSIP_EXTERNALENDPOINT=peer0.org1.example.com:7051
+	- FABRIC_LOGGING_SPEC=debug
+	## the following setting redirects chaincode container logs to the peer container logs
+	- CORE_VM_DOCKER_ATTACHSTDOUT=true
+	- CORE_PEER_LOCALMSPID=Org1MSP
+	- CORE_PEER_MSPCONFIGPATH=/etc/hyperledger/msp/peer
+	##
+	- CORE_PEER_TLS_ENABLED=true
+	- CORE_PEER_TLS_CLIENTAUTHREQUIRED=true
+	- CORE_PEER_TLS_KEY_FILE=/etc/hyperledger/msp/peer/tls/key.pem
+	- CORE_PEER_TLS_CERT_FILE=/etc/hyperledger/msp/peer/tls/cert.pem
+	- CORE_PEER_TLS_ROOTCERT_FILE=/etc/hyperledger/msp/peer/cacerts/org1.example.com-cert.pem
+	- CORE_PEER_TLS_CLIENTROOTCAS_FILES=/etc/hyperledger/msp/peer/cacerts/org1.example.com-cert.pem
+	# # the following setting starts chaincode containers on the same
+	# # bridge network as the peers
+	# # https://docs.docker.com/compose/networking/
+	- CORE_VM_DOCKER_HOSTCONFIG_NETWORKMODE=fixtures_default
+  working_dir: /opt/gopath/src/github.com/hyperledger/fabric
+  command: peer node start
+  ports:
+	- 7051:7051 // notice internal and external are the same
+  volumes:
+	  - /var/run/:/host/var/run/
+	  - ./channel/crypto-config/peerOrganizations/org1.example.com/peers/peer0.org1.example.com/:/etc/hyperledger/msp/peer
+  depends_on:
+	- orderer.example.com
+```
+
+### using chaincode to chaincode calls and collections
+The peer's discovery service will required information
```

Comment:
...discovery service will require**d** information...
remove the 'd'

Reply:
fixed